### PR TITLE
Add "Note" after various "SHOULD" rules in the spec

### PR DESCRIPTION
**Pull request by benjie**
I've not covered all of them, but hopefully enough to be helpful.

Part of #191

### CODE REVIEW

File: spec/GraphQLOverHTTP.md [hunk 1]
```diff
@@ -236,9 +239,17 @@ A client SHOULD indicate the media types that it supports in responses using the
 `Accept` HTTP header as specified in
 [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231).
 
+Note: If a client does not supply the `Accept` header then the server may
+respond with an error, or with any content type it chooses. To ensure you client
```
Comment (ghmcadams) [inline spec/GraphQLOverHTTP.md:243]: Typo?

```suggestion
respond with an error, or with any content type it chooses. To ensure your client
```

File: spec/GraphQLOverHTTP.md [hunk 2]
```diff
@@ -313,12 +324,18 @@ type (as indicated by the `Content-Type` header).
 If the client does not supply a `Content-Type` header with a POST request, the
 server SHOULD reject the request using the appropriate `4xx` status code.
 
+Note: This is left as a "should" to allow your server to choose a different
+behavior should it wish to.
```
Comment (ghmcadams) [inline spec/GraphQLOverHTTP.md:328]: Is this needed?  As worded, it is no more than a casual definition of `SHOULD`.  If the intent is to reiterate and remind readers that a server can choose to vary behavior when a `SHOULD` is used, consider something like this:

```suggestion
Note: Rejecting such requests encourages clients to supply a `Content-Type` header with every POST request.  A server has the option to assume any media type they wish when none is supplied, with the understanding that parsing the request may fail.
```

File: spec/GraphQLOverHTTP.md [hunk 3]
```diff
@@ -313,12 +324,18 @@ type (as indicated by the `Content-Type` header).
...
+
 A server MAY support POST requests encoded with and/or accepting other media
 types.
 
 If a client does not know the media types the server supports then it SHOULD
 encode the request body in JSON (i.e. with `Content-Type: application/json`).
 
+Note: Request encoding with media type `application/json` is supported by every
+compliant _server_.
```
Comment (ghmcadams) [inline spec/GraphQLOverHTTP.md:337]: minor: My opinion is that italicizing keywords should be used always or never.

File: spec/GraphQLOverHTTP.md [hunk 4]
```diff
@@ -455,6 +472,9 @@ depending on the concrete error condition.
 
 Note: Typically this will be the `400` (Bad Request) status code.
 
+Note: This rule is "should" to maintain compatibility with legacy servers which
+may return 200 status codes even when this type of error occurs.
```
Comment (ghmcadams) [inline spec/GraphQLOverHTTP.md:476]: This note implies a changing normative rule that doesn't exist.  Consider rewording.

File: spec/GraphQLOverHTTP.md (hunk 3 above)
Comment (benjie) [reply spec/GraphQLOverHTTP.md:337]: The italics actually serves to turn the word into a [spec-md reference](https://spec-md.com/#sec-Definition-References), in this case clarifying that "server" definitely refers to "GraphQL-over-HTTP compliant server" or whatever we defined it to be.

Comment (benjie) [reply spec/GraphQLOverHTTP.md:337]: (But I would be perfectly okay with every appropriate mention of client/server being a reference.)

File: spec/GraphQLOverHTTP.md (hunk 4 above)
Comment (benjie) [inline spec/GraphQLOverHTTP.md:476]:
```suggestion
can return 200 status codes even when this type of error occurs.
```

File: spec/GraphQLOverHTTP.md [hunk 5]
```diff
@@ -455,6 +474,9 @@ depending on the concrete error condition.
 
 Note: Typically this will be the `400` (Bad Request) status code.
 
+Note: This rule is "should" to maintain compatibility with legacy servers which
```
Comment (spawnia) [inline spec/GraphQLOverHTTP.md:477]:
```suggestion
Note: This rule is "SHOULD" to maintain compatibility with legacy servers which
```

File: spec/GraphQLOverHTTP.md [hunk 6]
```diff
@@ -166,6 +166,9 @@ The GraphQL specification allows for many
 Servers and clients MUST support JSON and MAY support other, additional
 serialization formats.
 
+Note: If you allow other media types, particularly on requests, please ensure
+that you are aware of the security consequences of doing so.
```
Comment (spawnia) [inline spec/GraphQLOverHTTP.md:170]:
```suggestion
Note: Allowing other media types, particularly on requests, can be insecure.
```

Comment (benjie) [reply spec/GraphQLOverHTTP.md:170]: It's more like "can be insecure if you don't take the necessary precautions/actions".

**Review by ghmcadams — APPROVED** [review]

**Review by spawnia — APPROVED** [review]

**Pull request merged**